### PR TITLE
feat [#2.0]: create patient

### PR DESCRIPTION
**Pull request by Aleksa1998**
Changing model, service and repository

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/DoctorService.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class DoctorService : AbstractUserService<DoctorUser>
         public DoctorService()
         {
             doctorRepository = new DoctorRepository(path);
-            patientsRepository = new PatientsRepository(path2);
+           // patientsRepository = new PatientsRepository(path2);
```
Comment (Maja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/DoctorService.cs:37]: Need to delete this line

File: PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs [hunk 2]
```diff
@@ -45,16 +45,16 @@ public void Remove(int appointmentid, int operationid)
...
-            Boolean hasOperationDoctor = doctorController.doesDoctorHaveAnOperationAtSpecificPerod(doctor, start, end,dateToString);
-            Boolean hasAppointmentPatient = patientController.doesPatientHaveAnAppointmentAtSpecificPeriod(start, end,dateToString, patient);
-            Boolean hasOperationPatient = patientController.doesPatientHaveAnOperationAtSpecificPeriod(start,end,dateToString, patient);
-            
-            if (hasAppointmentDoctor || hasAppointmentPatient || hasOperationDoctor || hasOperationPatient) return true;
+            Boolean hasAppointmentDoctor = doctorController.doesDoctorHaveAnAppointmentAtSpecificPeriod(doctor, start, end, dateToString);
+            Boolean hasOperationDoctor = doctorController.doesDoctorHaveAnOperationAtSpecificPerod(doctor, start, end, dateToString);
+            //Boolean hasAppointmentPatient = patientController.doesPatientHaveAnAppointmentAtSpecificPeriod(start, end,dateToString, patient);
+            //Boolean hasOperationPatient = patientController.doesPatientHaveAnOperationAtSpecificPeriod(start,end,dateToString, patient);
```
Comment (Maja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs:55]: These lines need to be deleted

File: PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs [hunk 3]
```diff
@@ -45,16 +45,16 @@ public void Remove(int appointmentid, int operationid)
...
+
+            if (hasAppointmentDoctor /*|| hasAppointmentPatient*/ || hasOperationDoctor /*|| hasOperationPatient*/) return true;
```
Comment (Maja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/OperationService.cs:57]: Delete comments in this statement

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 4]
```diff
@@ -139,10 +139,10 @@ public Boolean isTermNotAvailable(DoctorUser doctor, TimeSpan time, String dateT
         {
             Boolean hasAppointmentDoctor = doctorController.doesDoctorHaveAnAppointmentAtSpecificTime(doctor, time, dateToString);
             Boolean hasOperationDoctor = doctorController.doesDoctorHaveAnOperationAtSpecificTime(doctor, time, dateToString);
-            Boolean hasAppointmentPatient = patientController.doesPatientHaveAnAppointmentAtSpecificTime(time, dateToString, patient);
-            Boolean hasOperationPatient = patientController.doesPatientHaveAnOperationAtSpecificTime(time, dateToString, patient);
+            //Boolean hasAppointmentPatient = patientController.doesPatientHaveAnAppointmentAtSpecificTime(time, dateToString, patient);
+            //Boolean hasOperationPatient = patientController.doesPatientHaveAnOperationAtSpecificTime(time, dateToString, patient);
```
Comment (Maja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:143]: Delete these lines

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 5]
```diff
@@ -139,10 +139,10 @@ public Boolean isTermNotAvailable(DoctorUser doctor, TimeSpan time, String dateT
...
 
-            if (hasAppointmentDoctor == true || hasAppointmentPatient == true || hasOperationDoctor == true || hasOperationPatient == true) return true;
+            if (hasAppointmentDoctor == true /*|| hasAppointmentPatient == true */|| hasOperationDoctor == true /*|| hasOperationPatient == true*/) return true;
```
Comment (Maja0505) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:145]: Delete comments in this statement

File: PatientWebApplication/PatientWebApplication/Controllers/PatientController.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        {
+
+            if (PatientService.Create(patientUser) == null)
+            {
+                return BadRequest();
+            }
+            else
+            {
+                return Ok();
+            }
```
Comment (Maja0505) [inline PatientWebApplication/PatientWebApplication/Controllers/PatientController.cs:34]: You can just return ok function, without else part

**Review by Maja0505 — CHANGES_REQUESTED** [review]
Delete all unnecessary commented parts of code

**Review by trki293 — CHANGES_REQUESTED** [review]
Needs to delete all parts of commented code

**Review by Maja0505 — APPROVED** [review]

**Review by trki293 — APPROVED** [review]
PR is approved.

**Pull request merged**